tsdb/compact_test.go

```diff
@@ -1845,3 +1845,152 @@ func TestCompactBlockMetas(t *testing.T) {
 	}
 	require.Equal(t, expected, output)
 }
+
+func TestDelayedCompaction(t *testing.T) {
```

Comment:
Tried to test many scenarios.

Will think about a test to make sure that operations that don't directly derive from the head compaction are not delayed (maybe I can just modify existing tests to add a case with the feature enabled.)

Reply:
added it.